Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -5,12 +5,81 @@ public enum GistMessageActions: String {
     case close = "gist://close"
 }
 
+/**
+ Class that implements the business logic for a inline message being displayed.
+ */
+class InlineMessageManager: MessageManager {}
+
+/**
+ Class that implements the business logic for a modal message being displayed.
+ */
+class ModalMessageManager: MessageManager {
+    private var messageLoaded = false
+    private var modalViewManager: ModalViewManager?
+    var messagePosition: MessagePosition = .top
+
+    override func routeLoaded(route: String) {
+        super.routeLoaded(route: route)
+
+        if route == currentMessage.messageId, !messageLoaded {
+            messageLoaded = true
+            if isMessageEmbed {
```

Comment:
`isMessageEmbed` is existing code in the codebase today, not code I wrote. So it may or may not be in reference to inline messaging. 

My intention with this PR was to create `InlineMessageManager` without modifying the logic used for Modals. 

We may be able to make more modifications to `ModalMessageManager` and `MessageManager`. Because of scope, I wanted to leave the Modal code alone in this change.

Reply:
embed messages in the existing code base meant to reference inline messages. So I would indeed work on different approach to reduce the complexity of the code branches

Reply:
I just pushed a commit that removed `isMessageEmbed` from the `ModalMessageManager` class as it was indeed related to inline messaging. This removal did increase the amount of code changes in this PR, but the amount of code change was small and it only removed unused code so I thought it was worth it.